File: src/main/java/no/stunor/origo/eventorapi/data/OrganisationRepository.java
package no.stunor.origo.eventorapi.data;

import org.springframework.stereotype.Repository;

import com.google.cloud.spring.data.firestore.FirestoreReactiveRepository;

import no.stunor.origo.eventorapi.model.organisation.Organisation;
import reactor.core.publisher.Mono;

@Repository
public interface OrganisationRepository extends FirestoreReactiveRepository<Organisation> {
    Mono<Organisation> findByOrganisationIdAndEventorId(String organisationId, String eventorId);
}

File: src/main/kotlin/no/stunor/origo/eventorapi/data/OrganisationRepository.kt
package no.stunor.origo.eventorapi.data

import no.stunor.origo.eventorapi.model.organisation.Organisation
import org.springframework.data.repository.CrudRepository
import org.springframework.stereotype.Repository
import java.util.UUID

@Repository
interface OrganisationRepository : CrudRepository<Organisation, UUID> {
    fun findByEventorRefAndEventorId(eventorRef: String, eventorId: String): Organisation?
}